recipes/automake/all/conanfile.py

```diff
@@ -1,140 +1,178 @@
-        self.env_info.PATH.append(bin_path)
-
-        bin_ext = ".exe" if self.settings.os == "Windows" else ""
-
-        aclocal = tools.unix_path(os.path.join(self.package_folder, "bin", "aclocal" + bin_ext))
-        self.output.info("Appending ACLOCAL environment variable with: {}".format(aclocal))
-        self.env_info.ACLOCAL.append(aclocal)
-
-        automake_datadir = tools.unix_path(self._datarootdir)
-        self.output.info("Setting AUTOMAKE_DATADIR to {}".format(automake_datadir))
-        self.env_info.AUTOMAKE_DATADIR = automake_datadir
-
-        automake_libdir = tools.unix_path(self._automake_libdir)
-        self.output.info("Setting AUTOMAKE_LIBDIR to {}".format(automake_libdir))
-        self.env_info.AUTOMAKE_LIBDIR = automake_libdir
-
-        automake_perllibdir = tools.unix_path(self._automake_libdir)
-        self.output.info("Setting AUTOMAKE_PERLLIBDIR to {}".format(automake_perllibdir))
-        self.env_info.AUTOMAKE_PERLLIBDIR = automake_perllibdir
-
-        automake = tools.unix_path(os.path.join(self.package_folder, "bin", "automake" + bin_ext))
-        self.output.info("Setting AUTOMAKE to {}".format(automake))
-        self.env_info.AUTOMAKE = automake
-
-        self.output.info("Append M4 include directories to AUTOMAKE_CONAN_INCLUDES environment variable")
-
-        self.user_info.compile = os.path.join(self._automake_libdir, "compile")
-        self.user_info.ar_lib = os.path.join(self._automake_libdir, "ar-lib")
+        bin_path = self.package_path.joinpath("bin")
+        self.output.info(f"Appending PATH environment variable: {bin_path}")
+        self.env_info.PATH.append(str(bin_path))
+
+        dataroot_path = self.package_path.joinpath("res")
+        self.output.info(f"Defining AUTOMAKE_DATADIR environment variable: {dataroot_path}")
+        self.env_info.AUTOMAKE_DATADIR = str(dataroot_path)
+        self.buildenv_info.define_path("AUTOMAKE_DATADIR", str(dataroot_path))
+
+        version = Version(self.version)
+        automake_dataroot_path = dataroot_path.joinpath(f"automake-{version.major}.{version.minor}")
+        self.output.info(f"Defining AUTOMAKE_LIBDIR environment variable: {automake_dataroot_path}")
+        self.env_info.AUTOMAKE_LIBDIR = str(automake_dataroot_path)
+        self.buildenv_info.define_path("AUTOMAKE_LIBDIR", str(automake_dataroot_path))
+
+        self.output.info(f"Defining AUTOMAKE_PERLLIBDIR environment variable: {automake_dataroot_path}")
+        self.env_info.AUTOMAKE_PERLLIBDIR = str(automake_dataroot_path)
+        self.buildenv_info.define_path("AUTOMAKE_PERLLIBDIR", str(automake_dataroot_path))
+
+        aclocal_bin = bin_path.joinpath("aclocal")
+        self.output.info(f"Defining ACLOCAL environment variable: {aclocal_bin}")
+        self.env_info.ACLOCAL = str(aclocal_bin)
+        self.buildenv_info.define_path("ACLOCAL", str(aclocal_bin))
+
+        aclocal_bin_conf_key = "tools.automake:aclocal"
```

Comment:
Same as the other PR, this will not work in v2

```suggestion
        aclocal_bin_conf_key = "user.automake:aclocal"
```